codec/address_test.go

```diff
@@ -39,3 +39,79 @@ func TestAddressJSON(t *testing.T) {
 	require.NoError(json.Unmarshal(addrJSONBytes, &parsedAddr))
 	require.Equal(addr, parsedAddr)
 }
+
+func TestEncodeWithChecksum(t *testing.T) {
+	require := require.New(t)
+
+	tests := []struct {
+		address     Address
+		checksumStr string
+	}{
+		{
+			address: CreateAddress(
+				byte(0),
+				ids.Empty,
+			),
+			checksumStr: "0x000000000000000000000000000000000000000000000000000000000000000000a7396ce9",
+		},
+		{
+			address: func() Address {
+				return CreateAddress(
+					byte(0),
+					[32]byte{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32},
+				)
+			}(),
+			checksumStr: "0x000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f20a10df6ab",
+		},
+		{
```

Comment:
What's the difference for this test case?

Reply:
This was a remanent of testing checksumming via EIP-55; removed that test case